homeassistant/components/blebox/cover.py

```diff
@@ -0,0 +1,109 @@
+"""BleBox cover entity."""
+
+import logging
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPENING,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    SUPPORT_STOP,
+    CoverDevice,
+)
+
+from . import BleBoxEntity, async_create_blebox_entities
+from .const import (
+    BLEBOX_TO_HASS_COVER_STATES,
+    BLEBOX_TO_HASS_DEVICE_CLASSES,
+    DOMAIN,
+    PRODUCT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add):
+    """Set up a BleBox entry."""
+
+    product = hass.data[DOMAIN][config_entry.entry_id][PRODUCT]
+    return await async_create_blebox_entities(
+        product, async_add, BleBoxCoverEntity, "covers"
+    )
+
+
+class BleBoxCoverEntity(BleBoxEntity, CoverDevice):
```

Comment:
We've renamed our entity classes.
```suggestion
class BleBoxCoverEntity(BleBoxEntity, CoverEntity):
```